test/integration/checkout_int_test.go

```diff
@@ -348,6 +349,29 @@ func (suite *CheckoutIntegrationTestSuite) TestChangeSummary() {
 	cp.ExpectExitCode(0)
 }
 
+func (suite *CheckoutIntegrationTestSuite) TestCheckoutFromArchive() {
+	suite.OnlyRunForTags(tagsuite.Checkout)
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	root := environment.GetRootPathUnsafe()
+	archive := filepath.Join(root, "test", "integration", "testdata", "checkout-from-archive", runtime.GOOS+".tar.gz")
+
+	cp := ts.SpawnWithOpts(
+		e2e.OptArgs("checkout", "org/project", "--from-archive", archive),
+		e2e.OptAppendEnv("HTTPS_PROXY=none://"), // simulate lack of network connection
+	)
+	cp.Expect("Checking out project: ActiveState-CLI/AlmostEmpty")
+	cp.Expect("Setting up the following dependencies:")
+	cp.Expect("└─ zlib@1.3.1")
+	cp.Expect("Sourcing Runtime")
+	cp.Expect("Unpacking")
+	cp.Expect("Installing")
+	cp.Expect("All dependencies have been installed and verified")
+	cp.Expect("Checked out project ActiveState-CLI/AlmostEmpty")
+	cp.ExpectExitCode(0)
```

Comment:
Let's also assert that the resulting runtime has some of the expected files. This test currently is just testing the UI/UX of the command itself, we should also verify that the end result is what we expect it to be.